src/Apps/Legion/Contexts/WizardNavigationContext/WizardNavigationContext.types.ts

```diff
@@ -46,6 +54,7 @@ export type WizardNavigationContextAction =
           type: WizardNavigationContextActionType.NAVIGATE_TO;
           payload: {
               stepNumber: number;
+              initialProps?: IAdditionalStepProps;
```

Comment:
this doesn't seem like the right approach from a types perspective. Now any step you send will accept these props but only a single one will do anything with the values which is confusing to a consumer. Can we find an alternative way to handle this?

Reply:
I think it is flexible enough right now with optional keys. So if a step requires any initial props to be set along with the step number we can extend the IAdditionalStepProps with optional keys. This way, for any step user can specify which additional props they want. Do you want me to create a mapping of additional props to the current step?

Reply:
I think the mapping makes sense. Something like
`payload: StepWithProps | StepWithoutProps` or `payload: StepPayload | ModifyStepPayload` and have `ModifyStepPayload` have stepNumber predefined to Modify step's number.

Reply:
used discriminating unions for the type here for this action